src/main/EndBoss.java

```diff
@@ -0,0 +1,125 @@
+package main;
+
+import interfaces.EnemyFishInterface;
+
+import java.util.Random;
+
+import javafx.scene.image.Image;
+
+
+/**
+ * An EndBoss is the end boss in this game which automatically moves over the screen,
+ * from left to right or right to left, field isLefty determines this. 
+ * If the PlayerFish collides with end boss without wielding the lance, 
+ * the PlayerFish dies. 
+ * If the PlayerFish collides with end boss with the lance, 
+ * end boss dies (it might take several hits).
+ * 
+ * @author Clinton Cao, Michiel Doesburg, Matthijs Halvemaan, Dmitry Malarev, Sunwei Wang.
+ */
+public class EndBoss extends Entity implements EnemyFishInterface {
+  private static String leftImageFileName = "SharkelliLeft.png";
+  private static String rightImageFileName = "SharekelliRight.png";
```

Comment:
You are using the wrong file name, this may cause an exception to be thrown! <review